### PR TITLE
`soci ztoc info` reports additional information

### PR DESCRIPTION
**Pull request by rdpsin**
Signed-off-by: Rishabh Singhvi <rdpsin@amazon.com>

*Issue #, if available:*

*Description of changes:*

*Testing performed:*

Example

```
% sudo ./soci ztoc info sha256:951fd618f3c6d61944b0fe2fe2c06ce7d43c6f0454d42f51000f37f31bb1e39e
{
  "version": "0.9",
  "build_tool": "AWS SOCI CLI v0.1",
  "span_size": 4000000,
  "max_span_id": 1,
  "num_files": 522,
  "num_multi_span_files": 1,
  "files": [
    {
      "filename": "bin/",
      "offset": 512,
      "size": 0,
      "type": "dir",
      "start_span": 0,
      "end_span": 0
    },
    {
      "filename": "bin/arch",
      "offset": 1024,
      "size": 0,
      "type": "symlink",
      "start_span": 0,
      "end_span": 0
    },
    {
      "filename": "bin/ash",
      "offset": 1536,
      "size": 0,
      "type": "symlink",
      "start_span": 0,
      "end_span": 0
    },
    {
      "filename": "bin/base64",
      "offset": 2048,
      "size": 0,
      "type": "symlink",
      "start_span": 0,
      "end_span": 0
    },
    {
      "filename": "bin/bbconfig",
      "offset": 2560,
      "size": 0,
      "type": "symlink",
      "start_span": 0,
      "end_span": 0
    },
    {
      "filename": "bin/busybox",
      "offset": 3072,
      "size": 841392,
      "type": "reg",
      "start_span": 0,
      "end_span": 0
    },
...
`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/soci/commands/ztoc/info.go [hunk 1]
```diff
@@ -30,15 +32,23 @@ import (
 )
 
 type Info struct {
-	Version   string     `json:"version"`
-	BuildTool string     `json:"build_tool"`
-	Files     []FileInfo `json:"files"`
+	Version           string             `json:"version"`
+	BuildTool         string             `json:"build_tool"`
+	Size              int64              `json:"size"`
```
Comment (djdongjin) [inline cmd/soci/commands/ztoc/info.go:37]: The `Size` is the artifact size right (ztoc blob size in db)?

**Review by djdongjin — COMMENTED** [review]
We should have a test for `ztoc info` to make sure the the output json can be unmarshalled back to the struct (`Info`, `FileInfo`). (IIRC, @turan18 is working on adding integration test for `ztoc info`)

File: cmd/soci/commands/ztoc/info.go (hunk 1 above)
Comment (rdpsin) [reply cmd/soci/commands/ztoc/info.go:37]: Yes.

**Comment by rdpsin** [comment]
Yes, I didn't include a test since I thought @turan18 was working on it. I can pick it up if no one is.

File: cmd/soci/commands/ztoc/info.go (hunk 1 above)
Comment (turan18) [reply cmd/soci/commands/ztoc/info.go:37]: Yeah I don't believe this change will break my integ test (unmarshal should just skip missing fields). I would have to update the current verification func which compares the fields values with the original ztoc.

**Comment by turan18** [comment]
> Yes, I didn't include a test since I thought @turan18 was working on it. I can pick it up if no one is.

I'm done with writing the test for it. I can publish a draft PR of it now and update it according to your changes

File: cmd/soci/commands/ztoc/info.go [hunk 2]
```diff
@@ -30,15 +32,23 @@ import (
...
+	SpanSize          compression.Offset `json:"span_size"`
+	MaxSpan           compression.SpanID `json:"max_span_id"`
```
Comment (Kern--) [inline cmd/soci/commands/ztoc/info.go:39]: Is the max span id something people care about? I think number of spans would be more useful.

**Review by djdongjin — APPROVED** [review]

**Comment by turan18** [comment]
If you could provide the issue link in the description that would autoclose the issue, otherwise LGTM

**Review by turan18 — APPROVED** [review]

**Pull request merged**